subprojects/core-api/src/main/java/org/gradle/api/artifacts/result/ComponentSelectionReason.java

```diff
@@ -56,6 +53,13 @@
      */
     boolean isExpected();
 
+    /**
+     * Informs whether the component is selected due to being a composite build.
+     *
+     * @since 4.5
+     */
+    boolean isCompositeBuild();
```

Comment:
This needs a better name. A selected dependency cannot be a composite build.

Suggestions:

1. `isCompositeParticipant`
2. `isLocallyBuilt`

I don't love either, but I think I prefer 1. 

@bigdaz what say you?